<!DOCTYPE HTML>
<html lang="en" class="light" dir="ltr">
    <head>
        <!-- Book generated using mdBook -->
        <meta charset="UTF-8">
        <title>Notes @ ETH Zürich</title>
        <meta name="robots" content="noindex">


        <!-- Custom HTML head -->
        
        <meta name="description" content="">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="theme-color" content="#ffffff">

        <link rel="icon" href="favicon.svg">
        <link rel="shortcut icon" href="favicon.png">
        <link rel="stylesheet" href="css/variables.css">
        <link rel="stylesheet" href="css/general.css">
        <link rel="stylesheet" href="css/chrome.css">
        <link rel="stylesheet" href="css/print.css" media="print">

        <!-- Fonts -->
        <link rel="stylesheet" href="FontAwesome/css/font-awesome.css">
        <link rel="stylesheet" href="fonts/fonts.css">

        <!-- Highlight.js Stylesheets -->
        <link rel="stylesheet" href="highlight.css">
        <link rel="stylesheet" href="tomorrow-night.css">
        <link rel="stylesheet" href="ayu-highlight.css">

        <!-- Custom theme stylesheets -->

        <!-- MathJax -->
        <script async src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    </head>
    <body class="sidebar-visible no-js">
    <div id="body-container">
        <!-- Provide site root to javascript -->
        <script>
            var path_to_root = "";
            var default_theme = window.matchMedia("(prefers-color-scheme: dark)").matches ? "navy" : "light";
        </script>

        <!-- Work around some values being stored in localStorage wrapped in quotes -->
        <script>
            try {
                var theme = localStorage.getItem('mdbook-theme');
                var sidebar = localStorage.getItem('mdbook-sidebar');

                if (theme.startsWith('"') && theme.endsWith('"')) {
                    localStorage.setItem('mdbook-theme', theme.slice(1, theme.length - 1));
                }

                if (sidebar.startsWith('"') && sidebar.endsWith('"')) {
                    localStorage.setItem('mdbook-sidebar', sidebar.slice(1, sidebar.length - 1));
                }
            } catch (e) { }
        </script>

        <!-- Set the theme before any content is loaded, prevents flash -->
        <script>
            var theme;
            try { theme = localStorage.getItem('mdbook-theme'); } catch(e) { }
            if (theme === null || theme === undefined) { theme = default_theme; }
            var html = document.querySelector('html');
            html.classList.remove('light')
            html.classList.add(theme);
            var body = document.querySelector('body');
            body.classList.remove('no-js')
            body.classList.add('js');
        </script>

        <input type="checkbox" id="sidebar-toggle-anchor" class="hidden">

        <!-- Hide / unhide sidebar before it is displayed -->
        <script>
            var body = document.querySelector('body');
            var sidebar = null;
            var sidebar_toggle = document.getElementById("sidebar-toggle-anchor");
            if (document.body.clientWidth >= 1080) {
                try { sidebar = localStorage.getItem('mdbook-sidebar'); } catch(e) { }
                sidebar = sidebar || 'visible';
            } else {
                sidebar = 'hidden';
            }
            sidebar_toggle.checked = sidebar === 'visible';
            body.classList.remove('sidebar-visible');
            body.classList.add("sidebar-" + sidebar);
        </script>

        <nav id="sidebar" class="sidebar" aria-label="Table of contents">
            <div class="sidebar-scrollbox">
                <ol class="chapter"><li class="chapter-item affix "><a href="index.html">Home</a></li><li class="spacer"></li><li class="chapter-item affix "><li class="part-title">22HS</li><li class="chapter-item "><div><strong aria-hidden="true">1.</strong> Computational Biology</div></li><li class="chapter-item "><div><strong aria-hidden="true">2.</strong> Data Mining I</div></li><li class="spacer"></li><li class="chapter-item affix "><li class="part-title">23FS + HS</li><li class="chapter-item "><a href="23fs/mham/mobile_health_and_activity_monitoring.html"><strong aria-hidden="true">3.</strong> Mobile Health and Activity Monitoring</a><a class="toggle"><div>❱</div></a></li><li><ol class="section"><li class="chapter-item "><div><strong aria-hidden="true">3.1.</strong> Introduction</div></li></ol></li><li class="chapter-item "><a href="23fs/fg/functional_genomics.html"><strong aria-hidden="true">4.</strong> Functional Genomics</a><a class="toggle"><div>❱</div></a></li><li><ol class="section"><li class="chapter-item "><a href="23fs/fg/01_modern_genomics_i.html"><strong aria-hidden="true">4.1.</strong> Modern Genomics I</a></li><li class="chapter-item "><a href="23fs/fg/02_modern_genomics_ii.html"><strong aria-hidden="true">4.2.</strong> Modern Genomics II</a></li><li class="chapter-item "><a href="23fs/fg/03_transcriptomics_i.html"><strong aria-hidden="true">4.3.</strong> Transcriptomics I</a></li><li class="chapter-item "><a href="23fs/fg/04_transcriptomics_ii.html"><strong aria-hidden="true">4.4.</strong> Transcriptomics II</a></li><li class="chapter-item "><div><strong aria-hidden="true">4.5.</strong> MicroRNAs and other small RNAs</div></li><li class="chapter-item "><div><strong aria-hidden="true">4.6.</strong> Proteomics</div></li><li class="chapter-item "><div><strong aria-hidden="true">4.7.</strong> Metabolomics</div></li><li class="chapter-item "><div><strong aria-hidden="true">4.8.</strong> Single Cell Mass Cytometry</div></li><li class="chapter-item "><div><strong aria-hidden="true">4.9.</strong> Protein Networks</div></li><li class="chapter-item "><div><strong aria-hidden="true">4.10.</strong> Epigenomics and Gene Regulation</div></li><li class="chapter-item "><div><strong aria-hidden="true">4.11.</strong> Qualitiy Control and Standards</div></li><li class="spacer"></li></ol></li><li class="chapter-item "><li class="part-title">24FS</li><li class="chapter-item "><div><strong aria-hidden="true">5.</strong> Biofluiddynamics</div></li><li class="chapter-item "><div><strong aria-hidden="true">6.</strong> Synthetic Biology I</div></li></ol>
            </div>
            <div id="sidebar-resize-handle" class="sidebar-resize-handle">
                <div class="sidebar-resize-indicator"></div>
            </div>
        </nav>

        <!-- Track and set sidebar scroll position -->
        <script>
            var sidebarScrollbox = document.querySelector('#sidebar .sidebar-scrollbox');
            sidebarScrollbox.addEventListener('click', function(e) {
                if (e.target.tagName === 'A') {
                    sessionStorage.setItem('sidebar-scroll', sidebarScrollbox.scrollTop);
                }
            }, { passive: true });
            var sidebarScrollTop = sessionStorage.getItem('sidebar-scroll');
            sessionStorage.removeItem('sidebar-scroll');
            if (sidebarScrollTop) {
                // preserve sidebar scroll position when navigating via links within sidebar
                sidebarScrollbox.scrollTop = sidebarScrollTop;
            } else {
                // scroll sidebar to current active section when navigating via "next/previous chapter" buttons
                var activeSection = document.querySelector('#sidebar .active');
                if (activeSection) {
                    activeSection.scrollIntoView({ block: 'center' });
                }
            }
        </script>

        <div id="page-wrapper" class="page-wrapper">

            <div class="page">
                                <div id="menu-bar-hover-placeholder"></div>
                <div id="menu-bar" class="menu-bar sticky">
                    <div class="left-buttons">
                        <label id="sidebar-toggle" class="icon-button" for="sidebar-toggle-anchor" title="Toggle Table of Contents" aria-label="Toggle Table of Contents" aria-controls="sidebar">
                            <i class="fa fa-bars"></i>
                        </label>
                        <button id="theme-toggle" class="icon-button" type="button" title="Change theme" aria-label="Change theme" aria-haspopup="true" aria-expanded="false" aria-controls="theme-list">
                            <i class="fa fa-paint-brush"></i>
                        </button>
                        <ul id="theme-list" class="theme-popup" aria-label="Themes" role="menu">
                            <li role="none"><button role="menuitem" class="theme" id="light">Light</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="rust">Rust</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="coal">Coal</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="navy">Navy</button></li>
                            <li role="none"><button role="menuitem" class="theme" id="ayu">Ayu</button></li>
                        </ul>
                        <button id="search-toggle" class="icon-button" type="button" title="Search. (Shortkey: s)" aria-label="Toggle Searchbar" aria-expanded="false" aria-keyshortcuts="S" aria-controls="searchbar">
                            <i class="fa fa-search"></i>
                        </button>
                    </div>

                    <h1 class="menu-title">Notes @ ETH Zürich</h1>

                    <div class="right-buttons">
                        <a href="print.html" title="Print this book" aria-label="Print this book">
                            <i id="print-button" class="fa fa-print"></i>
                        </a>

                    </div>
                </div>

                <div id="search-wrapper" class="hidden">
                    <form id="searchbar-outer" class="searchbar-outer">
                        <input type="search" id="searchbar" name="searchbar" placeholder="Search this book ..." aria-controls="searchresults-outer" aria-describedby="searchresults-header">
                    </form>
                    <div id="searchresults-outer" class="searchresults-outer hidden">
                        <div id="searchresults-header" class="searchresults-header"></div>
                        <ul id="searchresults">
                        </ul>
                    </div>
                </div>

                <!-- Apply ARIA attributes after the sidebar and the sidebar toggle button are added to the DOM -->
                <script>
                    document.getElementById('sidebar-toggle').setAttribute('aria-expanded', sidebar === 'visible');
                    document.getElementById('sidebar').setAttribute('aria-hidden', sidebar !== 'visible');
                    Array.from(document.querySelectorAll('#sidebar a')).forEach(function(link) {
                        link.setAttribute('tabIndex', sidebar === 'visible' ? 0 : -1);
                    });
                </script>

                <div id="content" class="content">
                    <main>
                        <h1 id="notes-taken--eth-zurich"><a class="header" href="#notes-taken--eth-zurich">Notes taken @ ETH Zurich</a></h1>
<h2 id="introduction"><a class="header" href="#introduction">Introduction</a></h2>
<p>Here I list all courses I took at ETH Zurich and provide a brief overview of the course content. The clickable links will lead you to the respective course notes.</p>
<p>Courses are organized by semester and year. The course notes are organized by lecture.</p>
<h3 id="22hs"><a class="header" href="#22hs">22HS</a></h3>
<ul>
<li>Computational Biology</li>
<li>Data Mining I</li>
<li>Information Systems for Engineers</li>
<li>Computational Systems Biology</li>
</ul>
<h3 id="23fs--hs"><a class="header" href="#23fs--hs">23FS + HS</a></h3>
<ul>
<li><a href="./23fs/mham/mobile_health_and_activity_monitoring.html">Mobile Health and Activity Monitoring</a></li>
<li><a href="./23fs/fg/functional_genomics.html">Functional Genomics</a></li>
<li>Introduction to Machine Learning</li>
<li>Big Data for Engineers</li>
<li>Statistical Models in Computational Biology</li>
</ul>
<p>No courses (except for seminar) taken in 23HS since I was doing an internship at Roche.</p>
<h3 id="24fs"><a class="header" href="#24fs">24FS</a></h3>
<ul>
<li>Biofluiddynamics</li>
<li>Synthetic Biology I</li>
</ul>
<div style="break-before: page; page-break-before: always;"></div><h1 id="mobile-health-and-activity-monitoring"><a class="header" href="#mobile-health-and-activity-monitoring">Mobile Health and Activity Monitoring</a></h1>
<div style="break-before: page; page-break-before: always;"></div><h1 id="functional-genomics"><a class="header" href="#functional-genomics">Functional Genomics</a></h1>
<p>A course collectively taught by multiple professors. Multiple-choice exam, lots of memorization.</p>
<h2 id="table-of-contents"><a class="header" href="#table-of-contents">Table of Contents</a></h2>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html">01. Modern Genomics I</a></li>
<li><a href="23fs/fg/02_modern_genomics_ii.html">02. Modern Genomics II</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html">03. Transcriptomics I</a></li>
<li><a href="23fs/fg/04_transcriptomics_ii.html">04. Transcriptomics II</a></li>
<li><a href="23fs/fg/05_mirnas_and_other_small_rnas.html">05. miRNAs and Other Small RNAs</a></li>
<li><a href="23fs/fg/06_proteomics.html">06. Proteomics</a></li>
<li><a href="23fs/fg/07_metabolomics.html">07. Metabolomics</a></li>
<li><a href="23fs/fg/08_single_cell_mass_cytometry.html">08. Single Cell Mass Cytometry</a></li>
<li><a href="23fs/fg/09_protein_networks.html">09. Protein Networks</a></li>
<li><a href="23fs/fg/10_epigenomics_and_gene_regulation.html">10. Epigenomics and Gene Regulation</a></li>
<li><a href="23fs/fg/11_quality_control_and_standards.html">11. Quality Control and Standards</a></li>
</ul>
<!-- toc -->
<div style="break-before: page; page-break-before: always;"></div><h1 id="modern-genomics-i"><a class="header" href="#modern-genomics-i">Modern Genomics I</a></h1>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#introduction">Introduction</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#timeline">Timeline</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#why-genomics">Why Genomics?</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#comparative-genomics-use-case-examples">Comparative genomics use case examples</a></li>
</ul>
</li>
<li><a href="23fs/fg/01_modern_genomics_i.html#classical-sequencing-method">Classical sequencing method</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#sanger-double-deoxy-sequencing">Sanger (double-deoxy) sequencing</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#automated-dye-sequencing">Automated Dye Sequencing</a></li>
</ul>
</li>
<li><a href="23fs/fg/01_modern_genomics_i.html#new-next-generation-sequencing-technologies">New (Next-generation) sequencing technologies</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#amplification-technologies">Amplification technologies</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#emulsion-pcr">Emulsion PCR</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#pcr-on-solid-support">PCR on solid support</a></li>
</ul>
</li>
<li><a href="23fs/fg/01_modern_genomics_i.html#barcoding-and-linked-reads">Barcoding and &quot;linked reads&quot;</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#sequencing-technologies">Sequencing technologies</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#pyrosequencing">Pyrosequencing</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#reversible-terminator-sequencing">Reversible terminator sequencing</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#sequencing-by-semi-conductor">Sequencing by semi-conductor</a></li>
</ul>
</li>
<li><a href="23fs/fg/01_modern_genomics_i.html#current-implementations-of-ngs">Current implementations of NGS</a></li>
</ul>
</li>
<li><a href="23fs/fg/01_modern_genomics_i.html#third-generation-sequencing-technologies">Third-generation sequencing technologies</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#environmental-sequencing">Environmental sequencing</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#how-to-deal-with-environmental-sequencing-data">How to deal with environmental sequencing data</a></li>
</ul>
</li>
<li><a href="23fs/fg/01_modern_genomics_i.html#single-cell-sequencing">Single-cell sequencing</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#why">Why?</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#how">How?</a>
<ul>
<li><a href="23fs/fg/01_modern_genomics_i.html#single-cell-isolation">Single-cell isolation</a></li>
<li><a href="23fs/fg/01_modern_genomics_i.html#whole-genome-amplification">Whole genome amplification</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="23fs/fg/01_modern_genomics_i.html#genomic-databases">Genomic databases</a></li>
</ul>
<h2 id="introduction-1"><a class="header" href="#introduction-1">Introduction</a></h2>
<p><span style="color:gray">Molecular biology is governed by the &quot;central dogma&quot;. Genomics is basically studying the <em>DNA</em> part of it.</span></p>
<p><strong>Genome sequence</strong>: complete listing of <u>all nucleotides</u> of <u>one organisms</u>, in correct order, and mapped to the chromosomes.</p>
<pre class="mermaid">graph LR
    A[DNA] --&gt;|Transcription| B[RNA]
    B --&gt;|Translation| C[Protein]
</pre>
<h3 id="timeline"><a class="header" href="#timeline">Timeline</a></h3>
<p>Efficient sequencing technology arrived rather late. Initially the sequencing process was cumbersome and radioactive.</p>
<ul>
<li>1975: &quot;dideoxy&quot; DNA sequencing (Sanger)</li>
<li>1977: first genome (<u>bacteriophage \( \phi X 174 \) </u>)</li>
<li>1995: first <u>cell</u> (Haemophilus influenzae)</li>
<li>1998: first animal (Caenorhabditis elegans)</li>
<li>2001: Homo sapiens</li>
<li>Today (February 2023)
<ul>
<li><span style="color:gray">genomes available for: 409,947 Bacteria, 4,988 Archaea, 47,200 Eukaryotes</span></li>
<li><span style="color:gray">human genomes fairly routine</span>
<ul>
<li><span style="color:gray">below 1000$ raw costs</span></li>
<li><span style="color:gray">&quot;Personal Genome Projects&quot; are enrolling 100’000s of volunteers, including their medical records</span></li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="why-genomics"><a class="header" href="#why-genomics">Why Genomics?</a></h3>
<ul>
<li>Because we want an inventory of all genes and functions</li>
<li>Because wea can compare genomes to learn about evolution, to get hints on gene function, etc
<ul>
<li>Comparison can be either based on DNA or protein</li>
<li>Alignments, dot plots, whole chromosome comparison</li>
</ul>
</li>
</ul>
<h3 id="comparative-genomics-use-case-examples"><a class="header" href="#comparative-genomics-use-case-examples">Comparative genomics use case examples</a></h3>
<ul>
<li>Gene prediction
<ul>
<li>Gene prediction algorithms that use homology (=comparative genomics result) information: SLAM, SGP, Twinscan (= N-SCAN)...</li>
<li><img src="23fs/fg/img/20240202163019.png" alt="UCSC genome browser gene prediction" /></li>
</ul>
</li>
<li>Gene family evolution</li>
</ul>
<h2 id="classical-sequencing-method"><a class="header" href="#classical-sequencing-method">Classical sequencing method</a></h2>
<h3 id="sanger-double-deoxy-sequencing"><a class="header" href="#sanger-double-deoxy-sequencing">Sanger (double-deoxy) sequencing</a></h3>
<p>Natural DNA extension requires 3'-OH. The dideoxy method uses a 2',3'-dideoxy nucleotide, which lacks the 3'-OH group. This causes the DNA chain to terminate. By introducing different dideoxy nucleotides, the sequence can be read.</p>
<h3 id="automated-dye-sequencing"><a class="header" href="#automated-dye-sequencing">Automated Dye Sequencing</a></h3>
<p>Variants of Sanger sequencing. Still utilize the dideoxy method to terminate DNA elongation. The difference is that the dideoxy nucleotides are labeled with different fluorescent dyes. The sequence is read by a laser.</p>
<p><img src="23fs/fg/img/20240202163700.png" alt="Two variants of Automated Dye Sequencing" /></p>
<p>Dye terminator sequencing is now widely used over the rather cumbersome (4 tubes per sample) dye primer chemistry.</p>
<h2 id="new-next-generation-sequencing-technologies"><a class="header" href="#new-next-generation-sequencing-technologies">New (Next-generation) sequencing technologies</a></h2>
<p>Generally involves first <strong>amplifying</strong> the DNA, then <strong>sequencing</strong> it. Sequencing is done by detecting the nucleotides as they are incorporated into the growing DNA strand (sequencing by synthesis). High-throughput is achieved by parallelizing the sequencing process.</p>
<h3 id="amplification-technologies"><a class="header" href="#amplification-technologies">Amplification technologies</a></h3>
<p>First-generation amplification technology: needs DNA-library in bacterial vectors --&gt; cumbersome and biased</p>
<p>Improvement: get rid of bacteria</p>
<h4 id="emulsion-pcr"><a class="header" href="#emulsion-pcr">Emulsion PCR</a></h4>
<p>Improvement: bacterium free, but still needs cloning</p>
<pre class="mermaid">%%{init: {&quot;graph&quot;: {&quot;htmlLabels&quot;: false}} }%%
graph TD
    A[&quot;`**Fragment** the DNA, ligate **adapters** to ends, make **single-stranded**`&quot;] --&gt; B[&quot;Attach to microbeads&quot;]
    B --&gt; C[&quot;`PCR-amplify, in a **water-oil emulsion**`&quot;]
    C --&gt; D[&quot;`Enrich beads having successful amplifications, then place into regular lattice
    (see figure below for details)`&quot;]
</pre>
<p><img src="23fs/fg/img/20240203111110.png" alt="Emulsion PCR enrichment step" /></p>
<p>In short, the enrichment is done by capturing the second (5'-end) primer of the PCR product onto a large polysyrene bead.</p>
<h4 id="pcr-on-solid-support"><a class="header" href="#pcr-on-solid-support">PCR on solid support</a></h4>
<p><img src="23fs/fg/img/20240203111302.png" alt="PCR on solid support" /></p>
<h3 id="barcoding-and-linked-reads"><a class="header" href="#barcoding-and-linked-reads">Barcoding and &quot;linked reads&quot;</a></h3>
<p><img src="23fs/fg/img/20240203114453.png" alt="Barcoding" /></p>
<p><img src="23fs/fg/img/20240203114512.png" alt="10X technology" /></p>
<h3 id="sequencing-technologies"><a class="header" href="#sequencing-technologies">Sequencing technologies</a></h3>
<p>First-generation sequencing needs DNA size-separation on a gel</p>
<p>Improvement: get rid of gel (sequencing by synthesis)</p>
<h4 id="pyrosequencing"><a class="header" href="#pyrosequencing">Pyrosequencing</a></h4>
<p><img src="23fs/fg/img/20240203113950.png" alt="Pyrosequencing" /></p>
<h4 id="reversible-terminator-sequencing"><a class="header" href="#reversible-terminator-sequencing">Reversible terminator sequencing</a></h4>
<p><img src="23fs/fg/img/20240203113926.png" alt="Reversible terminator sequencing" /></p>
<h4 id="sequencing-by-semi-conductor"><a class="header" href="#sequencing-by-semi-conductor">Sequencing by semi-conductor</a></h4>
<p>Directly detects the release of H+ ions when a nucleotide is incorporated into the growing DNA strand.</p>
<p><img src="23fs/fg/img/20240203114721.png" alt="Sequencing by semi-conductor" /></p>
<h3 id="current-implementations-of-ngs"><a class="header" href="#current-implementations-of-ngs">Current implementations of NGS</a></h3>
<ul>
<li>Illumina
<ul>
<li>Illumina NovaSeq 6000</li>
<li>PCR on <u>solid support</u></li>
<li><u>reversible terminator</u> sequencing</li>
<li>read length ca. 250bp</li>
<li><code>1e14</code> bp per run</li>
</ul>
</li>
<li>Ion Torrent / Life Techn. Inc
<ul>
<li>Ion Gene Studio S5</li>
<li>PCR on <u>beads</u></li>
<li>sequencing by <u>semi-conductor</u></li>
<li>read length ca. 600bp</li>
<li><code>1e10</code> bp per run</li>
</ul>
</li>
</ul>
<h2 id="third-generation-sequencing-technologies"><a class="header" href="#third-generation-sequencing-technologies">Third-generation sequencing technologies</a></h2>
<p><strong>Single molecule sequencing</strong>. <strong>No</strong> need for <strong>amplification</strong>.</p>
<p>Characterized by extremely long reads, but also high error rates.</p>
<ul>
<li>Pacific Biosciences
<ul>
<li><strong>SMRT</strong> (single molecule real time) sequencing</li>
<li><img src="23fs/fg/img/20240203114758.png" alt="pacbio1" /></li>
<li><img src="23fs/fg/img/20240203114809.png" alt="pacbio1" /></li>
</ul>
</li>
<li>Oxford <strong>Nanopore</strong>
<ul>
<li>MinION</li>
<li><img src="23fs/fg/img/20240203114829.png" alt="oxford nanopore" /></li>
</ul>
</li>
</ul>
<blockquote>
<p><strong>Self-note</strong>: minimap2 is a popular aligner for long reads.</p>
</blockquote>
<h2 id="environmental-sequencing"><a class="header" href="#environmental-sequencing">Environmental sequencing</a></h2>
<p><u>Traditional genome sequencing</u> requires individual <strong>cell isolation</strong> and <strong>cultivation</strong>. This is not possible for the majority of microorganisms. But one advantage is that it's possible to re-assemble the genome from the reads.</p>
<p>Environmental sequencing: directly sequence DNA extracted from the environment without purification and clonal cultivation. Genome assembly is generally not possible.</p>
<blockquote>
<p><strong>Self-note</strong>: data generated from environmental sequencing is typically large in size, but highly fragmented and contaminated. Lots of exciting research in this area.</p>
</blockquote>
<h3 id="how-to-deal-with-environmental-sequencing-data"><a class="header" href="#how-to-deal-with-environmental-sequencing-data">How to deal with environmental sequencing data</a></h3>
<ul>
<li>Novel gene discovery
<ul>
<li>Sequence identity comparison to known genes
<ul>
<li>But &gt; 50% of the environmental genomes are not similar to any known genome</li>
</ul>
</li>
</ul>
</li>
<li>Novel gene families</li>
<li>Gene family clustering (similar samples have similar gene family distribution)</li>
</ul>
<h2 id="single-cell-sequencing"><a class="header" href="#single-cell-sequencing">Single-cell sequencing</a></h2>
<h3 id="why"><a class="header" href="#why">Why?</a></h3>
<ul>
<li><strong>Heterogeneity</strong> in cell populations
<ul>
<li>Tumor cells</li>
<li>Immune cells</li>
<li>Microbial communities</li>
<li>Developmental biology</li>
</ul>
</li>
</ul>
<h3 id="how"><a class="header" href="#how">How?</a></h3>
<p>In short, we first get <strong>single</strong> cells, then amplify the <strong>whole genome</strong> and sequence it.</p>
<p>The challenges lie in the bolded parts.</p>
<h4 id="single-cell-isolation"><a class="header" href="#single-cell-isolation">Single-cell isolation</a></h4>
<p>(In the very first &quot;single cell&quot; genomics paper, the &quot;single&quot; cells were literally picked manually...nowadays we don't do that)</p>
<ol>
<li>Sorting with optical tweezers
<img src="23fs/fg/img/20240304231454.png" alt="Optic tweezers" /></li>
<li>Dilution series</li>
<li>Flow sorting
<img src="23fs/fg/img/20240304231416.png" alt="Flow sorting" /></li>
</ol>
<h4 id="whole-genome-amplification"><a class="header" href="#whole-genome-amplification">Whole genome amplification</a></h4>
<p>Steps summarized:</p>
<ol>
<li>MDA</li>
<li>phi 29 debranching</li>
<li>S1 nuclease digestion</li>
<li>DNA pol I nick translation</li>
<li>Cloning</li>
</ol>
<ul>
<li>Isothermal <strong>Multiple displacement amplification</strong> (MDA)
<ul>
<li><strong>Phi29</strong> DNA polymerase</li>
<li><strong>Random primers</strong></li>
<li><strong>Isothermal</strong> amplification</li>
</ul>
</li>
</ul>
<p><img src="23fs/fg/img/20240304231641.png" alt="MDA" /></p>
<p>After MDA, we obtained a &quot;<strong>hyperbranched chromosome</strong>&quot;. After <em>debranching</em> and cloning, we can sequence and re-assemble the genome.</p>
<p><img src="23fs/fg/img/20240304231833.png" alt="Next steps" /></p>
<p>The debranching is done by incubating phi 29 DNA pol with hyperbranched DNA <strong>without any primer</strong>. The <em>strand-replacement</em> activity of phi 29 DNA pol will remove the hyperbranched structure.</p>
<p>S1 nucleases are used to remove the remaining single-stranded DNA.</p>
<p>Nicks are filled in by DNA pol I (has 5'-&gt;3' exonuclease activity).</p>
<h2 id="genomic-databases"><a class="header" href="#genomic-databases">Genomic databases</a></h2>
<p>This section likely won't be covered in the exam.</p>
<blockquote>
<p><strong>General popular resources</strong>:</p>
<ul>
<li>Raw data: NCBI <strong>sequence read archive (SRA)</strong> (also it's European counterpart, EBI <strong>European Nucleotide Archive (ENA)</strong>, but they are basically the same thing now)
<ul>
<li>seq quality score included</li>
<li>but incomplete: legacy &amp; newer data not available</li>
<li>gigantic in size</li>
</ul>
</li>
<li>Sequencing projects: <a href="https://gold.jgi.doe.gov/">GOLD (Genomes OnLine Database)</a>
<ul>
<li>keep track of &quot;who is sequencing what&quot; and responsible researchers (contacts), funding sources, sequencing centers etc</li>
</ul>
</li>
<li>Genome browsers
<ul>
<li>Display features (genes, transcripts...) on the genomes, show annotations (conflicts, variants also included), homolog search</li>
<li>UCSC genome browser, Ensembl (popular in Europe)</li>
<li>Pros and cons of genome browsers
<ul>
<li>Pros
<ul>
<li>easy to use</li>
<li>regularly updated</li>
<li>automated annotation pipelines =&gt; fast to include new genomes</li>
<li>very powerful export utilities (<code>BioMart</code> in Ensembl)</li>
<li>API for local access</li>
<li>DAS (distributed annotation system) for data exchange</li>
<li>long term project, stable funding, likely not going away</li>
</ul>
</li>
<li>Cons
<ul>
<li>focus on vertebrates, few other genomes</li>
<li>complex db schema</li>
<li>popular, so can be slow</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>Special ones:</p>
<ul>
<li>Comparative genomics databases
<ul>
<li>STRING (protein-protein interactions, focused on microbial genomes, maintained by von Mering group at UZH)</li>
<li>specialized on comparing genomes (at nucleotide-level, or gene-level)</li>
<li>to visualize evidence of selection (exons, regulatory sites, ...)</li>
<li>to infer past evolution of genomes (rearrangements, gains, losses, ...)</li>
<li>to establish gene histories (orthology, paralogy, synteny, ...)</li>
<li>often require extensive offline computation before they go online</li>
<li>some of their services also offered by generic genome browsers/sites.</li>
</ul>
</li>
<li>Organism-specific databases
<ul>
<li>Flybase, Wormbase, TAIR, SGD...</li>
<li>community driven, extensive manual input</li>
<li>specific terms, abbreviations, gene names...</li>
</ul>
</li>
<li>Specialized databases
<ul>
<li>IGSR: human population genetics</li>
<li>OMIM: known disease-causing mutations</li>
<li>KEGG: metabolic pathways and enzymes</li>
</ul>
</li>
</ul>
</blockquote>
<div style="break-before: page; page-break-before: always;"></div><h1 id="modern-genomics-ii"><a class="header" href="#modern-genomics-ii">Modern Genomics II</a></h1>
<p>I cannot understand his accent...</p>
<ul>
<li><a href="23fs/fg/02_modern_genomics_ii.html#contents-that-i-think-are-important">Contents that (I think) are important</a>
<ul>
<li><a href="23fs/fg/02_modern_genomics_ii.html#linkage-disequilibrium-ld">Linkage disequilibrium (LD)</a>
<ul>
<li><a href="23fs/fg/02_modern_genomics_ii.html#coefficient-of-linkage-disequilibrium--d-">Coefficient of linkage disequilibrium ( D )</a></li>
</ul>
</li>
<li><a href="23fs/fg/02_modern_genomics_ii.html#relative-measures-of-ld">Relative measures of LD</a></li>
<li><a href="23fs/fg/02_modern_genomics_ii.html#nucleotide-diversity-pi">Nucleotide diversity (\pi)</a></li>
</ul>
</li>
<li><a href="23fs/fg/02_modern_genomics_ii.html#course-contents-according-to-recording">Course contents according to recording</a></li>
</ul>
<h2 id="contents-that-i-think-are-important"><a class="header" href="#contents-that-i-think-are-important">Contents that (I think) are important</a></h2>
<h3 id="linkage-disequilibrium-ld"><a class="header" href="#linkage-disequilibrium-ld">Linkage disequilibrium (LD)</a></h3>
<h4 id="coefficient-of-linkage-disequilibrium--d-"><a class="header" href="#coefficient-of-linkage-disequilibrium--d-">Coefficient of linkage disequilibrium \( D \)</a></h4>
<p>Given two loci \(A\) and \(B\) (can have allele \(A\) or \(a\), \(B\) or \(b\)), the coefficient of linkage disequilibrium \(D\) is defined as</p>
<p>\[
D = \frac{P_{AB} - P_A P_B}{P_A P_B}
\]</p>
<p>Below shows two examples of \(D\) calculation.</p>
<p><img src="23fs/fg/./img/02-ld-example1.png" alt="LD example 1" /></p>
<p>\[
D = \frac{P_{AB} - P_{A} P_{B}}{P_{A} P_{B}} = \frac{5}{12} - \frac{8}{12}\times\frac{6}{12} = \frac{1}{12}
\]</p>
<p>A and B are linked to some extent.</p>
<p><img src="23fs/fg/./img/02-ld-example2.png" alt="LD example 2" /></p>
<p>\[
D = \frac{P_{AB} - P_{A} P_{B}}{P_{A} P_{B}} = \frac{6}{12} - \frac{8}{12}\times\frac{9}{12} = 0
\]</p>
<p>Here A and B are not linked at all.</p>
<p>Basically we are testing whether P(A) and P(B) are <u>independent</u>.</p>
<h3 id="relative-measures-of-ld"><a class="header" href="#relative-measures-of-ld">Relative measures of LD</a></h3>
<ol>
<li>\(r^2 = \frac{D}{P_A P_a P_B P_b}\)</li>
<li>\(D' = \frac{D}{D_{\max}}\) if \(D\) is positive, \(D' = \frac{D}{D_{\min}}\) if \(D\) is negative</li>
</ol>
<h3 id="nucleotide-diversity-pi"><a class="header" href="#nucleotide-diversity-pi">Nucleotide diversity \(\pi\)</a></h3>
<p><span style="color:gray">Average proportion of pairwise differences between the sequences</span></p>
<p>Give a multiple sequence alignment, \(\pi\) is defined as</p>
<p>\[
\pi = \sum_{i &lt; j} \frac{\pi_{ij}}{n_c}
\]</p>
<p>where \(n_c = \frac{n(n-1)}{2}\) is the number of pairwise comparisons, and \(\pi_{ij}\) is the proportion of differences between the \(i\)th and \(j\)th sequences.</p>
<p>Below shows an example of \(\pi\) calculation.</p>
<p><img src="23fs/fg/./img/02-nt-diversity.png" alt="Nucleotide diversity example" /></p>
<p>\[
\pi = \sum_{i &lt; j} \frac{\pi_{ij}}{n_c} = \frac{\pi_{12} + \pi_{13} + \pi_{23}}{3} = 0.2
\]</p>
<p>Basically just calculating the average number of differences in the matrix.</p>
<h2 id="course-contents-according-to-recording"><a class="header" href="#course-contents-according-to-recording">Course contents according to recording</a></h2>
<p>Title: Applications of next-generation sequencing and other genomic techniques</p>
<ol>
<li>Resequencing to identify genetic basis of phenotypic variation
<ol>
<li>Outlier approach</li>
<li>Genome-wide association studies (<strong>GWAS</strong>) and <strong>linkage disequilibrium</strong></li>
</ol>
</li>
<li>Introduction to RNA-seq (transcriptome) focusing on genome duplication</li>
<li>Resequencing to detect signature of selection: <strong>nucleotide diversity</strong></li>
</ol>
<p>Microbial community and metagenomics, ChIP-seq, epigenomics, whole genome assembly as reference</p>
<ul>
<li>Evolutionary functional genomics can...
<ul>
<li>identify ecologically relevant genes</li>
<li>inferring selection &amp; population processes</li>
</ul>
</li>
</ul>
<div style="break-before: page; page-break-before: always;"></div><h1 id="transcriptomics-i"><a class="header" href="#transcriptomics-i">Transcriptomics I</a></h1>
<p>Unimportant text is grayed out but worth reading for fluency and context.</p>
<ul>
<li><a href="23fs/fg/03_transcriptomics_i.html#motivation">Motivation</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#rna-species">RNA species</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#rna-selection-methods">RNA selection methods</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#rna-seq-experiment-workflow-overview">RNA-seq experiment workflow overview</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#rna-seq-experiment-design">RNA-seq experiment design</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#rna-seq-read-mapping-methods">RNA-seq read mapping methods</a>
<ul>
<li><a href="23fs/fg/03_transcriptomics_i.html#considerations">Considerations</a>
<ul>
<li><a href="23fs/fg/03_transcriptomics_i.html#how-to-map-reads-to-a-genome">How to map reads to a genome?</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#what-if-we-want-to-map-billions-of-reads">What if we want to map billions of reads?</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#how-to-deal-with-snps-and-indels">How to deal with SNPs and indels?</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#how-to-deal-with-introns-basically-very-large-insertions">How to deal with introns (basically very large insertions)?</a></li>
</ul>
</li>
<li><a href="23fs/fg/03_transcriptomics_i.html#bowtie">Bowtie</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#tophat2">TopHat2</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#star">STAR</a></li>
</ul>
</li>
<li><a href="23fs/fg/03_transcriptomics_i.html#quality-control-and-reporting-considerations-after-mapping">Quality control and reporting considerations after mapping</a>
<ul>
<li><a href="23fs/fg/03_transcriptomics_i.html#phread-quality">PHREAD quality</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#read-trimming">Read trimming</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#multiple-alignments-and-report-considerations">Multiple alignments and report considerations</a></li>
</ul>
</li>
<li><a href="23fs/fg/03_transcriptomics_i.html#expression-quantification">Expression quantification</a>
<ul>
<li><a href="23fs/fg/03_transcriptomics_i.html#how-to-count-reads">How to count reads?</a>
<ul>
<li><a href="23fs/fg/03_transcriptomics_i.html#em-estimation">EM estimation</a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#alignment-free-rna-seq-quantification">Alignment-free RNA-seq quantification</a></li>
</ul>
</li>
<li><a href="23fs/fg/03_transcriptomics_i.html#normalization">Normalization</a></li>
</ul>
</li>
</ul>
<h2 id="motivation"><a class="header" href="#motivation">Motivation</a></h2>
<p><span style="color:gray"> We are interested in the expression of genes in a
cell. Altough <u>proteins are the actors</u> of genes instead of mRNA
transcripts, <em>mRNAs transcripts are easier to measure</em> because selection
has chosen them to convey the genetic message in a faithful way. Also there are
plenty of interesting <em>non-coding RNAs</em>. </span></p>
<h2 id="rna-species"><a class="header" href="#rna-species">RNA species</a></h2>
<p>80% of the total RNA is rRNA. 14% is tRNA. Only 1-3% is mRNA. All other species
account for the remaining 1-3%.</p>
<p><img src="23fs/fg/img/03-rna-species.png" alt="RNA species" style="zoom:50%;" /><img
src="img/03-rna-prevalence-pie-chart.png" alt="Prevalence of RNA species"
style="zoom:50%;" /></p>
<h2 id="rna-selection-methods"><a class="header" href="#rna-selection-methods">RNA selection methods</a></h2>
<p><img src="23fs/fg/img/03-rna-selection-method.png" alt="RNA selection methods" /></p>
<ul>
<li>A: total RNA
<ul>
<li>Unbiased estimation</li>
<li>Dominate by rRNA, not very informative</li>
</ul>
</li>
<li>B: ribosomal reduction/depletion
<ul>
<li>Use probes that bind to rRNA and pull them out</li>
<li>&lt; 5% rRNA left</li>
</ul>
</li>
<li>C: poly-A selection
<ul>
<li>Use probes that bind to poly-A tails</li>
<li>Selects for mRNA, and also some long non-coding RNAs (because they also have
poly-A tails)</li>
</ul>
</li>
<li>D: cDNA capture
<ul>
<li>Design target probes that bind to a certain set of RNAs</li>
<li>Limited transcriptome coverage</li>
</ul>
</li>
</ul>
<p><span style="color:gray"> 90% of transcriptome research utilize poly-A
selection; 5% of them go for ribosomal depletion because their mRNAs are of low
quality and fragmented. The rest 5% do have high-quality mRNAs but still go for
ribosomal depletion because they want to estimate other RNA species at the same
time.</span></p>
<p>Note: current illumina sequencers cannot sequence miRNAs (20-25nt) and mRNA
fragments (100-300nt) together (<u>because of the large length discrepency</u>)</p>
<h2 id="rna-seq-experiment-workflow-overview"><a class="header" href="#rna-seq-experiment-workflow-overview">RNA-seq experiment workflow overview</a></h2>
<p>A typical RNA-seq experiment workflow is shown below.</p>
<p><img src="23fs/fg/img/03-rna-seq-experiment-workflow.png" alt="Typical RNA-seq experiment" /></p>
<p><span style="color:gray"> We can do either single-read sequencing or paired-end
sequencing (as is shown in the figure above) in the third step. </span></p>
<p><img src="23fs/fg/img/03-paired-end-seq.png" alt="Paired-end sequencing"
style="zoom:60%;" /></p>
<p>It can also be summarized into a flow diagram.</p>
<p><img src="23fs/fg/img/03-ran-seq-workflow.png" alt="RNA-seq workflow diagram"
style="zoom:67%;" /></p>
<p>The following sections will go into more details about each step.</p>
<h2 id="rna-seq-experiment-design"><a class="header" href="#rna-seq-experiment-design">RNA-seq experiment design</a></h2>
<ul>
<li>RNA-seq experiments are implemented as <strong>comparative experiments</strong></li>
<li>RNA-seq is used to measure the <strong>relative abundance (expression)</strong> of each
mRNA (gene) in a <u>sample</u>
<ul>
<li><span style="color:gray"><u>sample</u></span><span style="color:gray">: We
cannot sequence all the RNA fragments. Instead, we randomly select about 20M
fragments as a representation of the total RNA obtained. Note that we can
sample multiple times. </span></li>
</ul>
</li>
<li><span style="color:gray">RNA-seq can link gene expression to</span>
<ul>
<li><span style="color:gray">genotype</span></li>
<li><span style="color:gray">phenotype</span></li>
<li><span style="color:gray">treatment response </span></li>
</ul>
</li>
</ul>
<h2 id="rna-seq-read-mapping-methods"><a class="header" href="#rna-seq-read-mapping-methods">RNA-seq read mapping methods</a></h2>
<h3 id="considerations"><a class="header" href="#considerations">Considerations</a></h3>
<p><span style="color:gray"> Ideally, the read mapping process is as simple as
<strong>sequencing the transcribed RNAs</strong> and then <strong>mapping the
sequences back to the genome</strong> to identify the gene expression. </span></p>
<p><span style="color:gray"> However, in reality, the process is much more
complicated because <u>current sequencers can only perform faithful sequencing
up to a couple of hundreds of bp (usually 200-800bp)</u>, and a typical mRNA is
2-3kb long. Thus, the mRNA needs to be first chopped into fragments and then
sequenced. When we do the mapping, we're mapping the chopped reads back to the
genome instead of the whole spliced sequences. The existence of <u>alternative
splicing (protein isoforms)</u> and <u>sequencing errors</u> makes the mapping
process even more complicated. </span></p>
<p><span style="color:gray"> While discussing the mapping process, those
limitations and problems should be kept in mind. </span></p>
<h4 id="how-to-map-reads-to-a-genome"><a class="header" href="#how-to-map-reads-to-a-genome">How to map reads to a genome?</a></h4>
<ul>
<li>Mapping algorithms must be
<ul>
<li>Fast
<ul>
<li>Thus usually find perfect or near-perfect (tolerating 1bp mismatch) match</li>
</ul>
</li>
<li>Able to handle SNPs, indels and sequencing errors</li>
</ul>
</li>
</ul>
<p>We can always perform sequence alignment (here needs local
alignment/<strong>Smith-Waterman alignment</strong>). Of course the score matrix should be
good to determine the correct alignment. But this is slow because it requires
quadratic efforts.</p>
<h4 id="what-if-we-want-to-map-billions-of-reads"><a class="header" href="#what-if-we-want-to-map-billions-of-reads">What if we want to map billions of reads?</a></h4>
<ul>
<li><strong>Indexing method</strong>
<ul>
<li>Reads are aligned by <strong>index lookup</strong> instead of $O(n^2)$ seq comparison</li>
<li>Usually hashed so the lookup time is much faster</li>
<li>If lookup fails, actual sequence comparison is performed (more details on
this in next chapter)</li>
<li>Major aligners use the <u><strong>Burrows-Wheeler transform (BWT)</strong></u> to index
the reference genome
<ul>
<li><span style="color:gray">Very small. Even for the human genome the index
fits into 3GB RAM. </span></li>
</ul>
</li>
</ul>
</li>
<li>Example aligners
<ul>
<li><a href="23fs/fg/03_transcriptomics_i.html#bowtie"><strong>bowtie</strong>, <strong>bowtie2</strong></a></li>
<li>BWA</li>
<li><a href="23fs/fg/03_transcriptomics_i.html#tophat2"><strong>TopHat2</strong></a></li>
<li><a href="23fs/fg/03_transcriptomics_i.html#star"><strong>STAR</strong></a></li>
<li>SOAP</li>
<li>...</li>
</ul>
</li>
</ul>
<h4 id="how-to-deal-with-snps-and-indels"><a class="header" href="#how-to-deal-with-snps-and-indels">How to deal with SNPs and indels?</a></h4>
<ul>
<li><strong>SNPs</strong>
<ul>
<li><span style="color:gray">Can always be handled by performing real
Smith-Waterman alignment but expensive </span></li>
<li>At mismatched positions, try all possible bases and search the BWT index
again
<ul>
<li>Computing effort grows exponentially</li>
</ul>
</li>
<li><strong>Gapped alignment</strong>
<ul>
<li>Chop reads into short segments (seeds)</li>
<li>align those seeds in a <u>mismatch-free</u> manner (typically using BWT
index lookup again)</li>
<li>stitch them back together</li>
<li>Might require multiple rounds of chopping</li>
</ul>
</li>
</ul>
</li>
<li><strong>Indels</strong>
<ul>
<li>Can only be found with <strong>gapped alignment</strong> (described above), BWT lookup is
not enough</li>
</ul>
</li>
</ul>
<h4 id="how-to-deal-with-introns-basically-very-large-insertions"><a class="header" href="#how-to-deal-with-introns-basically-very-large-insertions">How to deal with introns (basically very large insertions)?</a></h4>
<ul>
<li>Approaches
<ul>
<li>Map directly to <strong>trapscript sequences</strong> (no intron) not to the genome
<ul>
<li>Pro: introns are not a problem</li>
<li>Cons: unknown genes/isoforms cannot be detected</li>
</ul>
</li>
<li><strong>Spliced alignment</strong> to genome
<ul>
<li>Pro: finds reads from unknown gene loci or unknown isoforms</li>
<li>Con: larger search space, potentially more false positives, wrong
alignment to <u>pseudogenes</u></li>
</ul>
</li>
<li>Combination of above</li>
</ul>
</li>
<li>Difference between spliced alignment and gapped alignment
<ul>
<li>Gapped alignment: longer gaps means high score penalty and lower score</li>
<li>Spliced alignment: long gaps at canonical splice-sites <u>are allowed</u>,
but lacking canonical splice-sites or gaps elsewhere (at non-canonical
splice sites) are penalized</li>
</ul>
</li>
<li>Pseudogenes
<ul>
<li>associated <em>retrotransposed</em> spliced sequence of a certain gene, thus
<em>intron-free</em>, and appears &quot;<em>exonic</em>&quot;</li>
<li>Have almost the same sequence as the original gene</li>
<li>Not functional because they do not have access to promotors so won't be
transcribed</li>
<li>Usually contain more mutations because there are no selection pressures to
keep them intact</li>
<li>Exonic reads will map to both the gene and its pseudogene, but likely
preferring gene placement due to lack of mutations</li>
<li>A spliced read could be incorrectly assigned to the pseudogene as it appears
to be exonic, preventing higher scoring spliced alignments from being
pursued</li>
</ul>
</li>
</ul>
<h3 id="bowtie"><a class="header" href="#bowtie">Bowtie</a></h3>
<p><img src="23fs/fg/img/03-bowtie.png" alt="Bowtie" /></p>
<ul>
<li>Step-by-step backward searching the <u><strong>suffix</strong></u> of the query sequence in
the BWT index
<ul>
<li>Exact matching</li>
</ul>
</li>
<li>Searched suffix appears consecutively in BWT</li>
<li>The size of range in BWT shrinks or remains the same</li>
<li><u><strong>almost does not handle mismatches</strong></u>; a single mismatch will lead to
<u>empty BWT</u> range / failed index lookup
<ul>
<li>mismatches can come from
<ul>
<li>Sequencing error (illumina: $1/1000$)</li>
<li>True variation (SNPs, human mutation rate $\approx 1/10^4$</li>
</ul>
</li>
<li>mismatches are not rare events! at least 10% of &gt;100nt reads</li>
</ul>
</li>
<li>Empty BWT range activates <strong>backtracking</strong>
<ul>
<li>All possible bases are tried at the mismatched position</li>
<li>Gapped alignment</li>
</ul>
</li>
</ul>
<h3 id="tophat2"><a class="header" href="#tophat2">TopHat2</a></h3>
<p><strong>Handles mismatches</strong> by gapped alignment. It <u>first</u> performs (optional)
<u>transcriptome alignment</u> (exonic), and then performs <u>genome
alignment</u> for the unmapped reads. The unmapped reads in the second step are
multi-exon spanning reads and undergo <u>spliced alignment</u> by chopping the
reads into even shorter segments and splice site identification etc.</p>
<blockquote>
<p>The figure is too complicated, not recommended to go into details but still
attached here for reference. <img src="23fs/fg/img/03-tophat2.png" alt="TopHat2 pipeline" /></p>
</blockquote>
<h3 id="star"><a class="header" href="#star">STAR</a></h3>
<p><strong>Direct</strong> alignment of non-contiguous sequences to the reference genome.</p>
<ul>
<li>First find Maximal Mappable Prefix (MMP) of the read
<ul>
<li>MMP is the longest prefix of the read that can be mapped to the reference
genome</li>
<li>MMP is found by searching the BWT index</li>
<li>Possible to tolerate a certain degree of mismatches using extend mode</li>
</ul>
</li>
<li>Stop extending MMP if MMP covers the whole read or encounters a mismatch</li>
<li>In the second case, find the MMP of the remaining suffix of the read</li>
<li>Repeat until the whole read is covered
<ul>
<li>Possible to use trim mode to get rid of A-tail or adapter sequence</li>
</ul>
</li>
</ul>
<img src="23fs/fg/img/03-star.png" alt="STAR" style="zoom:67%;" />
<h2 id="quality-control-and-reporting-considerations-after-mapping"><a class="header" href="#quality-control-and-reporting-considerations-after-mapping">Quality control and reporting considerations after mapping</a></h2>
<h3 id="phread-quality"><a class="header" href="#phread-quality">PHREAD quality</a></h3>
<ul>
<li>Each called base is given a quality score $Q$</li>
<li>$Q = -10 \log_{10}(p)$ (PHRED score)
<ul>
<li>$p$ is the estimated probability of the base being called incorrectly</li>
<li>$Q$ is the negative log of the probability
<ul>
<li>Q-score = 10: prob = 0.1</li>
<li>Q-score = 20: prob = 0.01</li>
<li>Q-score = 30: prob = 0.001</li>
</ul>
</li>
<li><strong>Q30</strong> is often used as a cutoff for high quality reads</li>
</ul>
</li>
<li>How to use: downweight the low-quality bases when computing the alignment
score</li>
<li><span style="color:gray">First introduced for capillary sequencers</span></li>
<li><span style="color:gray">PHRED scores are determined by the sequencer that
directly rates how reliable the measured signal is</span></li>
</ul>
<h3 id="read-trimming"><a class="header" href="#read-trimming">Read trimming</a></h3>
<p>Not all parts of the read are useful for alignment. It is common to trim the
reads.</p>
<ul>
<li>Reasons for trimming:
<ul>
<li>Systemetic errors of sequencer
<ul>
<li>Illumina sequencers have high error rate at the first few bases</li>
<li>Basically all sequencers have increasing error rate towards the end of the
read</li>
</ul>
</li>
<li>Adapter trimming
<ul>
<li>If the inserted DNA/RNA fragment is too short, the read will contain part
of the adapter (since the sequencer generally will sequence 100bp at each
end)</li>
<li>Can be challenging if:
<ul>
<li>The inserted sequence is 90-100bp -- the incorporated adapter sequence
will only be a few bp long</li>
<li>The read has many sequencing errors</li>
</ul>
</li>
</ul>
</li>
<li>3'-bias / degradation
<ul>
<li>The 3' end of the read is generally more likely to be degraded than the 5'
end</li>
<li>Depending on RNA degradation and extration protocols reads may not be
equally distributed along the transcripts</li>
</ul>
</li>
</ul>
</li>
<li>Trimming methods
<ul>
<li><strong>Hard trimming</strong>: trim a fixed number of bases from the beginning and/or
end</li>
<li><strong>Quality trimming</strong>: cut the end of the read as soon as the base quality
drops below a threshold</li>
<li><strong>Soft trimming</strong>: trim the read that cannot be aligned to the reference
genome</li>
</ul>
</li>
</ul>
<h3 id="multiple-alignments-and-report-considerations"><a class="header" href="#multiple-alignments-and-report-considerations">Multiple alignments and report considerations</a></h3>
<ul>
<li>A read may have multiple valid alignments with identical or similarly good
alignment scores
<ul>
<li>Aligners may use different reporting strategies:
<ul>
<li><strong>Randomly</strong> select one alignment from the top-scoring alignments</li>
<li>Report <u>all</u> alignments that are <u>within delta</u> of the
top-scoring alignment; clip if more than $N_{max}$ alignments are found</li>
<li>Report only alignments if they are <strong>unique</strong> (no other alignment within
delta of the alignment score)</li>
<li>Do not report anything if more than $N_{max}$ valid alignments are found</li>
<li>…</li>
</ul>
</li>
</ul>
</li>
<li>Whether a read has a <strong>unique alignment</strong> depends on
<ul>
<li>the read sequence and the sequence homology of the organism</li>
<li>the search algorithm of the aligner</li>
</ul>
</li>
<li>Whether a read has a <strong>single reported alignment</strong> depends <em>additionally</em> on
<ul>
<li>the reporting options strategy of the aligner</li>
</ul>
</li>
</ul>
<h2 id="expression-quantification"><a class="header" href="#expression-quantification">Expression quantification</a></h2>
<ul>
<li>Expression quantification on
<ul>
<li><strong>Gene level</strong>
<ul>
<li>Reads belong to a gene locus</li>
</ul>
</li>
<li><strong>Isoform level</strong>
<ul>
<li>Reads belong to an isoform</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="how-to-count-reads"><a class="header" href="#how-to-count-reads">How to count reads?</a></h3>
<ul>
<li><strong>Unique mapped reads</strong>
<ul>
<li>Multi-reads are ignored – potentially biased quantification</li>
</ul>
</li>
<li><strong>All mapped reads</strong>
<ul>
<li>Assignment of multi-reads need abundance estimation of overlapped
genes/isoforms</li>
</ul>
</li>
</ul>
<p><img src="23fs/fg/img/03-read-counting.png" alt="Read counting" /></p>
<h4 id="em-estimation"><a class="header" href="#em-estimation">EM estimation</a></h4>
<ul>
<li>General steps
<ol>
<li>Estimate abundances based on <u>uniquely mapping reads only</u></li>
<li>For each multi-read, divide it between the transcripts to which it maps, proportionally to their abundances estimated in the first step</li>
<li>Recompute abundances based on updated counts for each transcript
r.</li>
<li>Continue with Step 2</li>
</ol>
</li>
<li>Model the sequencing and analysis process with a likelihood function</li>
<li>E-step: Compute expected read counts given current expression levels</li>
<li>M-step: Compute expression values maximizing likelihood given expected read counts</li>
</ul>
<h4 id="alignment-free-rna-seq-quantification"><a class="header" href="#alignment-free-rna-seq-quantification">Alignment-free RNA-seq quantification</a></h4>
<p>Instead of building BWT, we build hashed k-mer index based on known
transcriptome, and search the k-mers in the reads to map and compute abundance.</p>
<p>Easier to understand through side-by-side comparison with the traditional
alignment dependent method:</p>
<p><img src="23fs/fg/img/03-kmer.png" alt="Alignment dependent v.s. alignment free" /></p>
<p>While allocating k-mer to transcript, we try to make each transcript covered
as uniformly as possible.</p>
<ul>
<li>Pros
<ul>
<li>Accurate and fast in quantifying <strong>KNOWN TRANSCRIPTS</strong></li>
<li>Counts can be aggregated to get gene-level quantification</li>
</ul>
</li>
<li>Cons
<ul>
<li>Less well annotated genomes – less accurate results</li>
<li>No alignments – no information about SNPs</li>
<li>Transcript coverage is not uniform</li>
</ul>
</li>
</ul>
<h3 id="normalization"><a class="header" href="#normalization">Normalization</a></h3>
<div style="break-before: page; page-break-before: always;"></div><h1 id="transcriptomics-ii"><a class="header" href="#transcriptomics-ii">Transcriptomics II</a></h1>
<ul>
<li><a href="23fs/fg/04_transcriptomics_ii.html#methods-of-exploratory-data-analysis">Methods of Exploratory Data Analysis</a></li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#clustering">Clustering</a>
<ul>
<li><a href="23fs/fg/04_transcriptomics_ii.html#distance-measures">Distance measures</a></li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#hierarchical-clustering">Hierarchical Clustering</a></li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#k-means-clustering">k-means Clustering</a></li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#comparison-of-hierarchical-clustering-and-k-means-clustering">Comparison of Hierarchical Clustering and k-means Clustering</a></li>
</ul>
</li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#dimensionality-reduction">Dimensionality Reduction</a>
<ul>
<li><a href="23fs/fg/04_transcriptomics_ii.html#principal-component-analysis-pca">Principal Component Analysis (PCA)</a></li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#t-sne">t-SNE</a></li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#umap">UMAP</a></li>
</ul>
</li>
<li><a href="23fs/fg/04_transcriptomics_ii.html#differential-expression-analysis">Differential Expression Analysis</a></li>
</ul>
<!-- ## Why exploratory data analysis?

tbc

## What do we study

- Good comparability: usually compare systems with small perturbations
  - Majority of the genes on the first diagonal
  - Only a few genes up- / downregulated

NEED FIGURE

- Bad comparability
  - TO FINISH

NEED FIGURE

- Meaningless comparison
  - Between different cell types

![Comparing adipocytes with astrocytes](img/04-20230313162850.png) -->
<h2 id="methods-of-exploratory-data-analysis"><a class="header" href="#methods-of-exploratory-data-analysis">Methods of Exploratory Data Analysis</a></h2>
<p>Typical shape of data we have in hand: a data matrix.</p>
<p><img src="23fs/fg/img/20240203134507.png" alt="Data matrix" /></p>
<p>where the <strong>columns</strong> are the <strong>samples</strong> and the <strong>rows</strong> are the <strong>features</strong> (usually gene expressions). (It can be the other way around, but this is the most common case.)</p>
<ul>
<li><strong>Clustering</strong>
<ul>
<li>Hierarchical clustering</li>
<li>k-means clustering</li>
</ul>
</li>
<li><strong>Dimensionality reduction</strong>
<ul>
<li>Matrix Factorization
<ul>
<li><strong>PCA</strong></li>
<li><strong>MDS</strong> (Multidimensional Scaling)</li>
</ul>
</li>
<li>Graph-based methods
<ul>
<li><strong>t-SNE</strong></li>
<li><strong>UMAP</strong></li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="clustering"><a class="header" href="#clustering">Clustering</a></h2>
<ul>
<li>Goals
<ul>
<li>group <strong>similar samples</strong>
<ul>
<li><span style="color:gray">evaluate similarity between expression profiles of the samples</span></li>
<li><span style="color:gray">test if similarities match the experimental design and effect sizes</span></li>
<li><span style="color:gray">test if variations within condition is smaller than between conditions</span></li>
<li><span style="color:gray">outliers detection</span></li>
</ul>
</li>
<li>group <strong>similar genes</strong>
<ul>
<li><strong>guilt by association</strong>: infer functions of unknown genes from known genes with the same expression pattern (co-expression)</li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="distance-measures"><a class="header" href="#distance-measures">Distance measures</a></h3>
<p>To cluster samples, we need to first define a &quot;distance&quot; measure between samples.</p>
<p>Commonly used distance measures:</p>
<ul>
<li><strong>Euclidean distance</strong> (typically for clustering <u>samples</u>)
<ul>
<li>Euclidean distance of two profiles $\mathbf{x}$ and $\mathbf{y}$ with $p$ genes (i.e. the distance between two $p$-dimensional vectors $\mathbf{x}$ and $\mathbf{y}$)</li>
<li>$d(\mathbf{x}, \mathbf{y}) = \sqrt{\sum_{i=1}^{p} \left(x_i - y_i\right)^2}$</li>
<li><strong>Expression values MUST BE LOG SCALE</strong></li>
</ul>
</li>
<li><strong>$1 - \text{corr}(\mathbf{x}, \mathbf{y})$</strong> (typically for clustering <u>genes</u>)
<ul>
<li>Correlation coefficient of two profiles $\mathbf{x}$ and $\mathbf{y}$ with $p$ samples</li>
<li>$\text{corr}(\mathbf{x}, \mathbf{y}) = \frac{\sum_{i=1}^{p}(x_i - \bar{x})(y_i - \bar{y})}{\sqrt{\sum_{i=1}^{p}(x_i - \bar{x})^2 \cdot \sum_{i-1}^{p}(y_i - \bar{y})^2}}$</li>
<li>$\bar{x} = \frac{1}{p}\sum_{i=1}^{p}x_i$</li>
<li>$\bar{y} = \frac{1}{p}\sum_{i=1}^{p}y_i$</li>
</ul>
</li>
</ul>
<h3 id="hierarchical-clustering"><a class="header" href="#hierarchical-clustering">Hierarchical Clustering</a></h3>
<p><strong>Algorithm</strong>:</p>
<ol>
<li>Compute the distance matrix ($n \times (n - 1) / 2 \rightarrow O(n^2)$) between all samples</li>
<li>Find pair with minimal distance and merge them</li>
<li>Update the distance matrix</li>
<li>Repeat 2-3 until all samples are merged</li>
</ol>
<p><strong>Parameters</strong>:</p>
<ul>
<li>distance measure for samples
<ul>
<li>usually $1 - \text{corr}(\mathbf{x}, \mathbf{y})$ for gene expression</li>
</ul>
</li>
<li>distance measure for clusters (<strong>linkage rule</strong>)
<ul>
<li><strong>Single</strong> linkage: <strong>minimum</strong> distance between any elements of the two clusters</li>
<li><strong>Complete</strong> linkage: <strong>maximum</strong> distance between any elements of the two clusters</li>
<li><strong>Average</strong> linkage: <strong>average</strong> distance between <em>all</em> elements of the two clusters</li>
<li><strong>Ward's</strong> linkage: <strong>minimal</strong> increase in <strong>intra-cluster variance</strong></li>
</ul>
</li>
</ul>
<p><strong>Input</strong>:</p>
<ul>
<li><strong>distance matrix</strong>
<ul>
<li>The linkage can be derived directly from the distance matrix.</li>
<li>Hence clustering algorithm only needs the distrance matrix as input, not the measurements individually.</li>
</ul>
</li>
</ul>
<h3 id="k-means-clustering"><a class="header" href="#k-means-clustering">k-means Clustering</a></h3>
<p><strong>Algorithm</strong>:</p>
<ol>
<li>Randomly assign each sample to one of the $k$ clusters</li>
<li>Compute the centroid (cluster center, average of the assigned samples) of each cluster</li>
<li>Assign each sample to the cluster with the closest centroid</li>
<li>Repeat 2-3 until convergence or a maximum number of iterations</li>
</ol>
<p><strong>Parameters</strong>:</p>
<ul>
<li>number of clusters $k$</li>
<li>distance measure for samples</li>
</ul>
<p><strong>Input</strong>:</p>
<ul>
<li>data matrix (cannot directly use distance matrix)</li>
</ul>
<p>This method <strong>minimizes the intra-cluster variance</strong>.</p>
<p>Choice of $k$ affects the result.</p>
<h3 id="comparison-of-hierarchical-clustering-and-k-means-clustering"><a class="header" href="#comparison-of-hierarchical-clustering-and-k-means-clustering">Comparison of Hierarchical Clustering and k-means Clustering</a></h3>
<div class="table-wrapper"><table><thead><tr><th></th><th>Hierarchical Clustering</th><th>k-means Clustering</th></tr></thead><tbody>
<tr><td>Computing time</td><td>$O(n^2 \log (n))$</td><td>$O(n \cdot k \cdot t)$ <ul><li>$t$: number of iterations</li><li>$k$: number of clusters</li></ul></td></tr>
<tr><td>Memory</td><td>$O(n^2)$</td><td>$O(n \cdot k)$</td></tr>
</tbody></table>
</div>
<p>When clustering large numbers of genes (&gt;<code>1e4</code>, hierarchical clustering is not practical</p>
<h2 id="dimensionality-reduction"><a class="header" href="#dimensionality-reduction">Dimensionality Reduction</a></h2>
<h3 id="principal-component-analysis-pca"><a class="header" href="#principal-component-analysis-pca">Principal Component Analysis (PCA)</a></h3>
<p><strong>Goal</strong>: Find a new coordinate system such that the first axis (principal component) captures the most variance, the second axis captures the second most variance, and so on.</p>
<p>The data is <strong>linearly</strong> transformed to a new coordinate system.</p>
<h3 id="t-sne"><a class="header" href="#t-sne">t-SNE</a></h3>
<p><strong>Algorithm</strong>:</p>
<ol>
<li>In the high-dimensional space, create a <em>probability distribution</em> that dictates how likely two points are to be neighbors.</li>
<li>Recreate a low dimensional space that follows the same probability distribution as best as possible.</li>
</ol>
<p>How to find the best low-dimensional representation:</p>
<ul>
<li><strong>preserve the pairwise distances</strong> between neighboring points in the high-dimensional space</li>
<li>non-linear, different transformations on <u>different regions</u></li>
</ul>
<p>Characteristic:</p>
<ul>
<li>Powerful, but need to fiddle with random seed and perplexity</li>
<li>Non-deterministic</li>
</ul>
<h3 id="umap"><a class="header" href="#umap">UMAP</a></h3>
<p>Uniform Manifold Approximation and Projection</p>
<p>Approach: Find for each point the neighbors and build simplices (simplex: a generalization of the concept of a triangle or tetrahedron to arbitrary dimensions) and then optimize the low-dimensional representation to preserve the simplices.</p>
<h2 id="differential-expression-analysis"><a class="header" href="#differential-expression-analysis">Differential Expression Analysis</a></h2>

                    </main>

                    <nav class="nav-wrapper" aria-label="Page navigation">
                        <!-- Mobile navigation buttons -->


                        <div style="clear: both"></div>
                    </nav>
                </div>
            </div>

            <nav class="nav-wide-wrapper" aria-label="Page navigation">

            </nav>

        </div>




        <script>
            window.playground_copyable = true;
        </script>


        <script src="elasticlunr.min.js"></script>
        <script src="mark.min.js"></script>
        <script src="searcher.js"></script>

        <script src="clipboard.min.js"></script>
        <script src="highlight.js"></script>
        <script src="book.js"></script>

        <!-- Custom JS scripts -->
        <script src="mermaid.min.js"></script>
        <script src="mermaid-init.js"></script>

        <script>
        window.addEventListener('load', function() {
            MathJax.Hub.Register.StartupHook('End', function() {
                window.setTimeout(window.print, 100);
            });
        });
        </script>

    </div>
    </body>
</html>
